File: src/main/java/com/everyonewaiter/domain/auth/JwtPayload.java
package com.everyonewaiter.domain.auth;

import static java.util.Objects.requireNonNull;

public record JwtPayload(String id, String subject) {

  public JwtPayload(Long id, String subject) {
    this(requireNonNull(id).toString(), subject);
  }

  public Long getLongId() throws NumberFormatException {
    return Long.parseLong(id);
  }

}

File: src/main/kotlin/com/everyonewaiter/common/jwt/JwtPayload.kt
package com.everyonewaiter.common.jwt

/**
 * JWT 생성 및 추출에 사용되는 데이터를 가진 클래스입니다.
 *
 * @property [id] JWT 고유 식별자
 * @property [subject] JWT 주체
 */
data class JwtPayload(
    val id: String,
    val subject: String,
)
